h/api/groups/logic.py

```diff
@@ -0,0 +1,30 @@
+from h.api import models
+
+
+def set_group_if_reply(annotation):
```

Comment:
I'm not sure if this is something that needs enforcing at all on the backend. The only thing that needs to be enforced is that an annotation must only be saved to a group for which the current request has write access.

I don't see a technical reason why a reply must be in the same group. Maybe the client should be responsible for copying the field from the parent the same way it copies the `uri` and creates `references` already.